file-formats/tabl/type/zif_aff_indx_v1.intf.abap

```diff
@@ -0,0 +1,42 @@
+INTERFACE zif_aff_indx_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Index Fields</p>
+    "! Index fields
+    BEGIN OF ty_indexfield,
+      "! <p class="shorttext">Field Name</p>
+      "! Field name
+      fieldname TYPE c LENGTH 30,
+      "! <p class="shorttext">Field Position</p>
+      "! Position
+      position  TYPE n LENGTH 4,
```

Comment:
Question: Must the field position be specified by the user or is this just calculated or just the position in the table?
If we used the order in the field list, we most probably would not be able to use a `sorted table`

Reply:
Very good question. The idea with the "..sorted table...with unique key field_name" was to ensure the uniqueness of each field name. But the index field order of course does not depend on the index field names. So the table cannot be ordered by field_name. 
The order of the index fields is very important. It is implicitly derived from the position of the fields in the index_fields table. 
-> I think we can get rid of the "position" field here as well. At least we do not like to see the "position" in a GUI editor.
-> Any ideas how we can ensure uniqueness of the index field names?

Reply:
> -> Any ideas how we can ensure uniqueness of the index field names?

What do you think about adding a secondary key to the internal table with a unique key field `field_name`
```
  TYPES
    tt_index_fields TYPE standard tABLE OF ty_indexfield WITH DEFAULT KEY
        WITH UNIQUE SORTED KEY name_key COMPONENTS field_name.
```

Adding a unique key to an internal table would mean that you cannot transform a JSON that has duplicate keys. The transformation will raise an exception